controllers/database_secret_handler.go

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2021 kloeckner.i GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import (
+	"context"
+	kciv1alpha1 "github.com/kloeckner-i/db-operator/api/v1alpha1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	DbSecretAnnotation = "db-operator/database"
+)
+
+/* ------ Secret Event Handler ------ */
+type secretEventHandler struct {
+	client.Client
+}
+
+func (e *secretEventHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+
+	logrus.Info("Secret Update Event processing started")
+
+	switch v := evt.ObjectNew.(type) {
+
+	default:
+		logrus.Error("Secret Update Event error! Unknown object: type=", v.GetObjectKind(), ", name=", evt.ObjectNew.GetNamespace(), "/", evt.ObjectNew.GetName())
+		return
+
+	case *corev1.Secret:
+		secretNew := evt.ObjectNew.(*corev1.Secret)
+		// find database object
+
+		// only annotated secrets are watched
+		annotations := secretNew.ObjectMeta.GetAnnotations()
+		dbcrName := annotations[DbSecretAnnotation]
+
+		if len(dbcrName) == 0 {
+			logrus.Error("Secret Update Event error! Annotation '", DbSecretAnnotation, "' value is empty.")
```

Comment:
think this check will throw a lot of false positive secret log erros. You should also check the second value for the map lookup.

```go
dbcrName, ok := annotations[DbSecretAnnotation]
if not ok {
  return # no secret to handle, annotation not exist
}

``` 

to distinct if the secret doen't have a annotation or the annotation was existing and empty.